daemon/src/main.rs

```diff
@@ -725,7 +734,81 @@ fn make_logger(quiet: bool) {
     .expect("Failed to initialize logger. Cancelling...");
 }
 
-fn is_daemon_running() -> Result<bool, String> {
+enum FdFindResult {
+    GotIt(u64),
+    NobodysListening,
+}
+
+fn find_fd_number(addr: &PathBuf) -> Result<FdFindResult, String> {
+    let sockets = std::path::PathBuf::from("/proc/net/unix");
+
+    let file = match std::fs::File::open(sockets) {
+        Ok(f) => f,
+        Err(e) => return Err(e.to_string()),
+    };
+
+    let reader = BufReader::new(file);
+
+    for line in reader.split(b'\n') {
+        let vec = match line {
+            Ok(v) => v,
+            Err(e) => return Err(e.to_string()),
+        };
+
+        let tabulated = vec.to_string_lossy();
+        let entries: Vec<&str> = tabulated.split(' ').collect();
+
+        // eight columns
+        // socket path is the last entry, socket number is the second to last entry
+        // ... unless there's no socket path.
+
+        if entries.len() < 8 {
+            continue;
+        }
+
+        let path = addr.to_string_lossy().to_string();
+        if *entries.last().unwrap() == path {
+            return match entries[entries.len() - 2].parse() {
+                Ok(value) => Ok(FdFindResult::GotIt(value)),
+                Err(_) => Err("Couldn't parse fd number".to_string()),
+            };
+        }
+    }
+
+    Ok(FdFindResult::NobodysListening)
+}
+
+fn check_fds(fd_num: u64, mut entry_dir: PathBuf) -> Result<bool, String> {
+    let name = format!("socket:[{}]", fd_num);
+    let linux_socket_num = PathBuf::from(name);
+
+    entry_dir.push("fd");
+
+    let entries = match entry_dir.read_dir() {
+        Ok(e) => e,
+        Err(e) => return Err(e.to_string()),
+    };
+
+    for entry in entries.flatten() {
+        match std::fs::read_link(entry.path()) {
+            Ok(real) => {
+                if real == linux_socket_num {
+                    return Ok(true);
+                }
+            }
+            Err(_) => continue,
+        };
+    }
+
+    Ok(false)
+}
+
+fn is_daemon_running(addr: &PathBuf) -> Result<bool, String> {
+    let fd_num = match find_fd_number(addr)? {
+        FdFindResult::GotIt(n) => n,
+        FdFindResult::NobodysListening => return Ok(false),
+    };
+
```

Comment:
I see you had to solve a different problem to get this work (the problem of detecting whether a daemon is already running). There's an easier way to do this. If the socket path already exists, simply try Pinging the daemon (like we do [here](https://github.com/LGFae/swww/blob/main/src/main.rs#L64)). If it returns, the daemon is already running. If it doesn't, we can continue.